src/raftstore/store/local_metrics.rs

```diff
@@ -0,0 +1,192 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use super::metrics::*;
+
+/// The buffered metrics counters for raft ready handling.
+#[derive(Debug, Default, Clone)]
+pub struct RaftReadyMetrics {
+    pub message: u64,
+    pub commit: u64,
+    pub append: u64,
+    pub snapshot: u64,
+}
+
+impl RaftReadyMetrics {
+    /// Flushes all metrics
+    fn flush(&mut self) {
+        // reset all buffered metrics once they have been added
+        if self.message > 0 {
+            STORE_RAFT_READY_COUNTER_VEC.with_label_values(&["message"])
+                .inc_by(self.message as f64)
+                .unwrap();
+            self.message = 0;
+        }
+        if self.commit > 0 {
+            STORE_RAFT_READY_COUNTER_VEC.with_label_values(&["commit"])
+                .inc_by(self.commit as f64)
+                .unwrap();
+            self.commit = 0;
+        }
+        if self.append > 0 {
+            STORE_RAFT_READY_COUNTER_VEC.with_label_values(&["append"])
+                .inc_by(self.append as f64)
+                .unwrap();
+            self.append = 0;
+        }
+        if self.snapshot > 0 {
+            STORE_RAFT_READY_COUNTER_VEC.with_label_values(&["snapshot"]).inc();
```

Comment:
I think use `HashMap<&str, u64>` will be simpler and more easy to prevent mistakes.